src/util/color_spaces.ts

```diff
@@ -27,111 +41,101 @@ const Xn = 0.950470, // D65 standard referent
+
+    const l = 116 * y - 16;
+    return [(l < 0) ? 0 : l, 500 * (x - y), 200 * (y - z), alpha];
 }
 
-function xyz2rgb(x: number) {
-    return 255 * (x <= 0.0031308 ? 12.92 * x : 1.055 * Math.pow(x, 1 / 2.4) - 0.055);
+function rgb2xyz(x: number): number {
+    return (x <= 0.04045) ? x / 12.92 : Math.pow((x + 0.055) / 1.055, 2.4);
 }
 
-function rgb2xyz(x: number) {
-    x /= 255;
-    return x <= 0.04045 ? x / 12.92 : Math.pow((x + 0.055) / 1.055, 2.4);
+function xyz2lab(t: number): number {
+    return (t > t3) ? Math.pow(t, 1 / 3) : t / t2 + t0;
 }
 
-// LAB
-function rgbToLab(rgbColor: Color): LABColor {
-    const b = rgb2xyz(rgbColor.r),
-        a = rgb2xyz(rgbColor.g),
-        l = rgb2xyz(rgbColor.b),
-        x = xyz2lab((0.4124564 * b + 0.3575761 * a + 0.1804375 * l) / Xn),
-        y = xyz2lab((0.2126729 * b + 0.7151522 * a + 0.0721750 * l) / Yn),
-        z = xyz2lab((0.0193339 * b + 0.1191920 * a + 0.9503041 * l) / Zn);
-
-    return {
-        l: 116 * y - 16,
-        a: 500 * (x - y),
-        b: 200 * (y - z),
-        alpha: rgbColor.a
-    };
-}
+// LAB -> sRGB
+function labToRgb([l, a, b, alpha]: LABColor): RGBColor {
+    let y = (l + 16) / 116,
+        x = isNaN(a) ? y : y + a / 500,
+        z = isNaN(b) ? y : y - b / 200;
 
-function labToRgb(labColor: LABColor): Color {
-    let y = (labColor.l + 16) / 116,
-        x = isNaN(labColor.a) ? y : y + labColor.a / 500,
-        z = isNaN(labColor.b) ? y : y - labColor.b / 200;
     y = Yn * lab2xyz(y);
     x = Xn * lab2xyz(x);
     z = Zn * lab2xyz(z);
-    return new Color(
+
+    return [
         xyz2rgb(3.2404542 * x - 1.5371385 * y - 0.4985314 * z), // D65 -> sRGB
         xyz2rgb(-0.9692660 * x + 1.8760108 * y + 0.0415560 * z),
         xyz2rgb(0.0556434 * x - 0.2040259 * y + 1.0572252 * z),
-        labColor.alpha
-    );
+        alpha,
+    ];
 }
 
-function interpolateLab(from: LABColor, to: LABColor, t: number) {
-    return {
-        l: interpolateNumber(from.l, to.l, t),
-        a: interpolateNumber(from.a, to.a, t),
-        b: interpolateNumber(from.b, to.b, t),
-        alpha: interpolateNumber(from.alpha, to.alpha, t)
-    };
+function xyz2rgb(x: number): number {
+    x = (x <= 0.00304) ? 12.92 * x : 1.055 * Math.pow(x, 1 / 2.4) - 0.055;
+    return (x < 0) ? 0 : (x > 1) ? 1 : x; // clip to 0..1 range
 }
 
-// HCL
-function rgbToHcl(rgbColor: Color): HCLColor {
-    const {l, a, b} = rgbToLab(rgbColor);
-    const h = Math.atan2(b, a) * rad2deg;
-    return {
-        h: h < 0 ? h + 360 : h,
-        c: Math.sqrt(a * a + b * b),
-        l,
-        alpha: rgbColor.a
-    };
+function lab2xyz(t: number): number {
+    return (t > t1) ? t * t * t : t2 * (t - t0);
 }
 
-function hclToRgb(hclColor: HCLColor): Color {
-    const h = hclColor.h * deg2rad,
-        c = hclColor.c,
-        l = hclColor.l;
-    return labToRgb({
-        l,
-        a: Math.cos(h) * c,
-        b: Math.sin(h) * c,
-        alpha: hclColor.alpha
-    });
+// sRGB -> HCL
```

Comment:
Regular jsdocs would be better here...

Reply:
These comments are just to separate the fragments related to a particular transformation. Like marking that the functions `rgb2xyz` and `xyz2lab` are subject to and used only by `rgbToLab`.
In addition, the exported `lab` `hcl` and `hsl` objects in the context of the file name and the names of the methods they contain are self-explanatory enough that additional comments are not necessary.

Reply:
If the comments are not needed, remove them, if they are needed turn them to jsdocs comments.
Separation comments are a code smell...

Reply:
Removed